File: backend/src/Autofix.Domain/Entities/Bookings/Booking.cs
using Autofix.Domain.Common;
using Autofix.Domain.Entities.People;
using Autofix.Domain.Entities.Vehicles;
using Autofix.Domain.Enum;

namespace Autofix.Domain.Entities.Booking;

public sealed class Booking : BaseEntity
{
    public Guid CustomerId { get; set; }
    public Customer? Customer { get; set; }
    public Guid VehicleId { get; set; }
    public Vehicle? Vehicle { get; set; }
    public Guid? BookingTimeSlotId { get; set; }
    public BookingTimeSlot? BookingTimeSlot { get; set; }
    public DateTime StartAt { get; set; }
    public DateTime EndAt { get; set; }
    public BookingStatus Status { get; set; } = BookingStatus.Created;
    public BookingPaymentOption PaymentOption { get; set; } = BookingPaymentOption.PayAtShop;
    public string Currency { get; set; } = "USD";
    public decimal Subtotal { get; set; }
    public decimal EstimatedLaborCost { get; set; }
    public decimal TaxAmount { get; set; }
    public decimal TotalEstimate { get; set; }
    public string? Notes { get; set; }
    public List<BookingServiceItem> Services { get; set; } = new();
}

File: frontend/src/features/dashboard/types/booking.ts
export type BookingStatus =
  | 'pending'
  | 'awaiting-approval'
  | 'approved'
  | 'in-progress'
  | 'changes-requested'
  | 'completed'
  | 'cancelled'

export interface BookingVehicle {
  id: string
  make: string
  model: string
  year: number
  trim: string
  engine: string
  plateNumber: string
  vin: string
  isDrivable: boolean
}

export interface BookingService {
  id: string
  name: string
  description: string
  estimatedDuration: string
  basePrice: number
  estimatedLaborCost: number
}

export interface BookingPricing {
  subtotal: number
  estimatedLaborCost: number
  taxAmount: number
  discountAmount: number
  totalBeforeDiscount: number
  totalEstimate: number
  currency: string
  loyaltyDiscountRate: number
  hasLoyaltyDiscount: boolean
}

export interface BookingEstimateWorkItem {
  id: string
  description: string
  laborHours: number
  hourlyRate: number
  lineTotal: number
}

export interface BookingEstimatePartItem {
  id: string
  partId: string
  partName: string
  quantity: number
  unitPrice: number
  availability: number
  lineTotal: number
}

export interface BookingEstimate {
  serviceOrderId: string
  status: number
  estimatedLaborCost: number
  estimatedPartsCost: number
  estimatedTotalCost: number
  discountAmount: number
  payableTotal: number
  loyaltyDiscountRate: number
  hasLoyaltyDiscount: boolean
  workItems: BookingEstimateWorkItem[]
  partItems: BookingEstimatePartItem[]
}

export interface Booking {
  id: string
  customerId: string
  vehicleId: string
  vehicle: BookingVehicle
  status: BookingStatus
  scheduledAt: string
  scheduledDate: string
  scheduledTime: string
  endAt: string
  estimatedCompletion?: string
  pricing: BookingPricing
  paymentOption: number
  services: BookingService[]
  estimate?: BookingEstimate
  notes?: string
  createdAt: string
  updatedAt: string
}

export interface BookingSummary {
  totalBookings: number
  active: number
  completed: number
  upcoming: number
}
